components/checkbox/overview.md

```diff
@@ -0,0 +1,275 @@
+* `Value` and `bind-Value`- mapped to the `Checked` property of the normal HTML checkbox
+  * The `Value` and `bind-Value` accept `bool` type
+* `Indeterminate` and `bind-Indeterminate` - mapped to the `Indeterminate` property of the normal HTML checkbox. Every time the state is changed (checked or unchecked) the `Indeterminate` is set to false.
+  * The `Indeterminate` and `bind-Indeterminate` accept `bool` type
+
+
+
+## Events
+* `ValueChanged` and `OnChange` - fire every time the Value gets changed.
+
+
+## Examples
+
+>caption One-way data binding by using Value and ValueChanged
+
+````CSHTML
+@*This example showcases one-way data binding by using Value and ValueChanged*@
+
+<h4 class="text-muted">Deliveries:</h4>
+
+@foreach (var delivery in Deliveries)
+{
+    <div>
+        <label class="text-muted">
+            <TelerikCheckBox Value="delivery.IsDelivered"
+                             ValueChanged="((bool value) => ChangeHandler(value, delivery.ProductName))" />
+            @delivery.ProductName
+        </label>
+    </div>
+}
+
+@if (AlreadyDelivered.Any())
+{
+<div>
+    <h6 class="text-info">Successfully delivered products:</h6>
+    <ul>
+        @{
+            foreach (var item in AlreadyDelivered)
+            {
+                <li>
+                    @item.ProductName
+                </li>
+            }
+        }
+    </ul>
+</div>   
+}
+
+@code {
+    public List<Delivery> Deliveries { get; set; }
+    public List<Delivery> AlreadyDelivered
+    {
+        get
+        {
+            return Deliveries.Where(x => x.IsDelivered == true).ToList();
+        }
+    }
+
+    void ChangeHandler(bool value, string productName)
+    {
+        var item = Deliveries.Where(x => x.ProductName == productName).First();
+        item.IsDelivered = value;
+    }
+
+    //In real case scenarios the model will be in a separate file.
+    public class Delivery
+    {
+        public string ProductName { get; set; }
+        public bool IsDelivered { get; set; }
+    }
+
+    //Generating dummy data
+    protected override void OnInitialized()
+    {
+        //Make your real data generation here.
+        Deliveries = new List<Delivery>();
+        Deliveries.Add(new Delivery()
+        {
+            ProductName = "PC",
+            IsDelivered = false
+        });
+        Deliveries.Add(new Delivery()
+        {
+            ProductName = "Mobile Phone",
+            IsDelivered = false
+        });
+        Deliveries.Add(new Delivery()
+        {
+            ProductName = "Headset",
+            IsDelivered = false
+        });
+        Deliveries.Add(new Delivery()
+        {
+            ProductName = "Monitor",
+            IsDelivered = false
+        });
+    }
+}
+````
+>caption The result from the code snippet above
+
+![screenshot to showcase checkbox with one-way data binding](images/one-way-data-binding-checkbox.jpg)
+
+>caption Observe the behavior of Indeterminate state
```

Comment:
The Indeterminate example, screesnshot, explaanations should be in a dedicated article